test/test_b2http.py

```diff
@@ -154,6 +154,45 @@ def test_too_many_requests_works_after_sleep(self):
             self.assertIs(self.response, _translate_and_retry(fcn, 3))
             self.assertEqual([call(2)], mock_time.mock_calls)
 
+    def test_too_many_requests_failed_after_sleep(self):
```

Comment:
add a comment: if we get a 429 response at the last retry attempt, requested delay should still be honored despite the fact that we won't make another attempt

Reply:
but that's not True in current implementation.